src/views/connectionUI.ts

```diff
@@ -84,19 +85,32 @@ export class ConnectionUI {
 	 * @param ignoreFocusOut Whether to ignoreFocusOut on the quickpick prompt
 	 * @returns The connection picked or created.
 	 */
-	public async promptForConnection(ignoreFocusOut = false): Promise<IConnectionInfo | undefined> {
+	public async promptForConnection(ignoreFocusOut: boolean = false): Promise<IConnectionInfo | undefined> {
 		let picklist = this._connectionStore.getPickListItems();
 		// We have recent connections - show them in a picklist
-		const selection = await this.promptItemChoice({
-			placeHolder: LocalizedConstants.recentConnectionsPlaceholder,
-			matchOnDescription: true,
-			ignoreFocusOut: ignoreFocusOut
-		}, picklist);
-		if (selection) {
-			return this.handleSelectedConnection(selection);
-		} else {
+		const connectionProfileQuickPick = await vscode.window.createQuickPick<IConnectionCredentialsQuickPickItem>();
+		connectionProfileQuickPick.items = picklist;
+		connectionProfileQuickPick.placeholder = LocalizedConstants.recentConnectionsPlaceholder;
+		connectionProfileQuickPick.matchOnDescription = true;
+		connectionProfileQuickPick.ignoreFocusOut = ignoreFocusOut;
+		connectionProfileQuickPick.canSelectMany = false;
+		connectionProfileQuickPick.busy = false;
+		connectionProfileQuickPick.show();
+		connectionProfileQuickPick.onDidChangeSelection(selection => {
+			if (selection[0]) {
+				// add progress notification and hide quickpick after user chooses an item from the quickpick
+				connectionProfileQuickPick.busy = true;
+				connectionProfileQuickPick.hide();
+				return this.handleSelectedConnection(selection[0]);
```

Comment:
@Charles-Gagnon one of the things you mentioned in a previous PR was that we should utilize the busy setting for when a user would like to test a connection. Right now the `this.handleSelectedConnection` returns back the connection if a user picks from a connection profile that is already saved shown below:
<img width="420" alt="Screen Shot 2022-06-02 at 5 42 23 PM" src="https://user-images.githubusercontent.com/23587151/171749753-b9ec8834-fec6-41f7-bf02-ba938cb9bcbd.png">

Was wondering how we would bring in the `Test Connection` part that would utilize the busy property of the quickpick. 
@chlafreniere & @dzsquared

Reply:
I'm not sure I understand what you're saying. Could you give an example of what you're talking about?

Reply:
I can discuss this more in a separate PR since it would be a further enhancement.
Context of the previous PR: https://github.com/microsoft/azuredatastudio/pull/19211#discussion_r858907784